R/runmodule.R

```diff
@@ -101,6 +101,31 @@ listModules <- function()
       gsub('^module\\.', '', . )
 }
 
+#' Population Data Module
+#'
+#' In mode = GETQ, returns query title. In mode = RUN, returns requested data.
+#'
+#' @keywords internal
+
+module.population <- function(var, mode, allqueries, aggkeys, aggfn, strtyr, endyr,
+                            filters, ounit)
+{
+    if(mode == GETQ) {
+        # As a procedure, should return title from query file or manual return?
+        # For more complex variables, will return multiple query titles.
+        'Population'
+    }
+    else {
+        print(paste('Function for processing variable', var))
+        population <- allqueries$'Population'; print(paste('Subset queries to ', var))
+        population$value <- aggfn(population$value); print('Aggregated')
+        population <- subset(population, Year >= strtyr & Year < endyr); print('Apply start and end years')
+        population <- filterfn(population, filters); print('Apply filters')
+        population <- unitconv(population, ounit); print('Convert units')
```

Comment:
Unlike `aggregate` and `filter`, we probably won't be able to do this one completely generically because converting units requires knowing _something_ about the kind of unit.  Probably we will have a bunch of functions like `unitconv_energy`, `unitconv_distance`, etc.  

So, a module like transportation would probably work by parsing the distance unit out of a string like "passenger-km" for both the units column of the table and the output unit.  Then it would call `unitconv_distance(tbl$value, gcam_distunit, output_distunit)` and assign the result back into the table.  

In the case of population, the units are counts (e.g., "thousands", "millions", etc.), so we need a `unitconv_counts`, but that's another feature for another branch.  What you should do right now is stub out `unitconv_counts` (put it in `units.R`).  It should just issue a warning that it is not yet implemented and return the table unmodified.